spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/ReactiveNeo4jTemplate.java

```diff
@@ -352,24 +351,36 @@ public ReactiveNeo4jTemplate(ReactiveNeo4jClient neo4jClient, Neo4jMappingContex
 												.setProperty(targetNodeDescription.getRequiredIdProperty(),
 													relatedInternalId);
 										}
-										Statement relationshipCreationQuery = statementBuilder
-											.createRelationshipCreationQuery(
-												neo4jPersistentEntity, relationship,
-												relatedValue instanceof Map.Entry ?
-													((Map.Entry<String, ?>) relatedValue).getKey() :
-													null,
-												relatedInternalId);
-
-										return
-											neo4jClient.query(renderer.render(relationshipCreationQuery))
-												.bind(fromId).to(FROM_ID_PARAMETER_NAME)
-												.run()
-												.then(processNestedAssociations(targetNodeDescription, valueToBeSaved));
-									})));
+
+										// handle creation of relationship depending on properties on relationship or not
+										RelationshipStatementHolder queryHolder = ctx.getRelationship()
```

Comment:
See imperative comments:
- variable naming
- better formatting of operator (e.g. use static imports for `RelationshipStatementHolder`)